engine/src/stack.cpp

```diff
@@ -3133,6 +3124,28 @@ bool MCStack::resolve_relative_path(MCStringRef p_path, MCStringRef& r_resolved)
     return false;
 }
 
+// PM-2015-01-26: [[ Bug 14435 ]] Make possible to set the filename using a relative path to the default folder
+bool MCStack::resolve_relative_path_to_default_folder(MCStringRef p_path, MCStringRef &r_resolved)
+{
+    if (MCStringIsEmpty(p_path))
+		return false;
+	
+	// If the relative path begins with "./" or ".\", we must remove this, otherwise
+    // certain system calls will get confused by the path.
+    uindex_t t_start_index;
+    MCAutoStringRef t_cur_dir;
+
+    if (MCStringBeginsWith(p_path, MCSTR("./"), kMCCompareExact)
+            || MCStringBeginsWith(p_path, MCSTR(".\\"), kMCCompareExact))
+        t_start_index = 2;
+    else
+        t_start_index = 0;
+	
+    MCS_getcurdir(&t_cur_dir);
+
+    return MCStringFormat(r_resolved, "%@/%*@", *t_cur_dir, MCRangeMake(t_start_index, MCStringGetLength(p_path) - t_start_index), p_path);
```

Comment:
The range argument here needs to be a pointer to an MCRange rather than an MCRange.